src/wayland-egldisplay.c

```diff
@@ -828,10 +829,9 @@ static bool getServerProtocolsInfo(struct wl_display *nativeDpy,
                 wl_display_roundtrip_queue(nativeDpy, queue);
                 zwp_linux_dmabuf_feedback_v1_destroy(default_feedback);
             }
-        } else if (protocols->wlDrm) {
-            wl_drm_add_listener(protocols->wlDrm, &drmListener, protocols);
-            wl_display_roundtrip_queue(nativeDpy, queue);
         }
+
+        /* Check that one of our two protocols provided the device name */
         result = protocols->drm_name != NULL;
```

Comment:
Do servers ever provide both protocols? I would have at least had to at some point to support a transition for clients. It looks like we'll leak at least one protocols->drm_name allocation in that case with this code.

Reply:
Yes they do, updated to free the existing drm_name before replacing it. This doesn't prioritize one source of the name over the other, it's just the last event to roll in. I think that should be fine, as these should be the same thing? This works with kwin since kwin sends a dev_t of zero through dmabuf and the path name through wl_drm, so we don't actually update the drm_name in our main_device handler as that devid doesn't exist. I think that should be fine?

Reply:
Yeah, I imagine it's fine to fix the buggy case. I was just curious what happens in the regular case where a valid dev_t is sent. I suppose they'd generally be the same, but it seems like we would want to prefer the dmabuf one if it were ever the case that they didn't match. This definitely seems like an improvement either way.